src/Workspaces/CoreTest/UtilityTest/SpecializedTasksTests.cs

```diff
@@ -17,13 +18,16 @@ public class SpecializedTasksTests
         [Fact]
         public void WhenAll_Null()
         {
+#pragma warning disable CA2012 // Use ValueTasks correctly (the instance is never created)
```

Comment:
Tag @stephentoub - something to be tuned in the analyzer, or seems fine?

Reply:
In this case the analyzer is behaving the way I expect. I could have written this test to call `AsTask()` on the result, but I felt it detracted from the test clarity as the call would not occur at runtime.

Reply:
Tests are special, in that they exercise things in ways you wouldn't actually do in production code, like doing things explicitly to get them to throw exceptions.  I'm actually surprised we run these analyzers over test code.

Reply:
As long as the noise level is low, I don't mind running analyzers in test code that could detect misuse that leads to flakiness. It's not uncommon for a test to synchronously get the result of an asynchronous operation because it's not concerned with thread pool usage.